src/Microsoft.Health.Blob/Features/Storage/BlobClientInitializer.cs

```diff
@@ -41,6 +42,12 @@ public BlobServiceClient CreateBlobClient(BlobDataStoreConfiguration configurati
             blobClientOptions.Retry.Delay = TimeSpan.FromSeconds(configuration.RequestOptions.ExponentialRetryBackoffDeltaInSeconds);
             blobClientOptions.Retry.NetworkTimeout = TimeSpan.FromMinutes(configuration.RequestOptions.ServerTimeoutInMinutes);
 
+            if (configuration.AuthenticationType == BlobDataStoreAuthenticationType.ManagedIdentity)
+            {
+                var defaultCredentials = new DefaultAzureCredential();
+                return new BlobServiceClient(new Uri(configuration.ConnectionString), defaultCredentials, blobClientOptions);
```

Comment:
FYI, Brandon raised this before, for managed identity, the ConnectionString is actually just the blob service url. For now it is okay to overload the same setting. We should document it in Dicom atleast to show how it looks for managed identity